### ISSUE TITLE
Script Generator: event estimation

### ISSUE THREAD

**Issue by JamesKingWork** [ISISComputingGroup/IBEX#6382]
As a script definition developer, I would like to be able to calculate and display how many events are in a script. Currently, we allow calculating of a time estimate, so for example, if we estimate 100 MEv then this is display as 1:40 MEv. This ticket is to allow more configuration of the estimation column. The column should allow setting of the time as a default which formats nicely for time or allow a script definition to define the column header and its own formatting. It may be best to do this as a general column time that is read only and is returned by the script definition, the details of the column heading/returned value formatting etc. can then be specified by the script definition.

## Acceptance criteria:

- [ ] Brief meeting is had to work out how to do a general solution to the requirement
- [ ] Decision is documented
- [ ] Estimation column defaults to time estimation
- [ ] I can configure the estimation column header from the script definition
- [ ] I can format the estimation column contents from the script definition
- [ ] I can configure how to calculate the total estimation

**Note:** Deemed "crucial" by Muon group

**Comment by JamesKingWork** [comment]
For prioritising from the IS: "it’s relatively high in our priorities because some users make a pretty long script with lots of lines in it. It would be very helpful to have the total number of events automatically calculated and displayed, so they can plan their beamtime."

**Comment by JamesKingWork** [comment]
+1 from MuSR

**Comment by DominicOram** [comment]
I think we should think about how to create general columns of read only things that the script returns to the user. The writer of the script definition should be able to create arbitrary ones of these.

**Comment by ThomasLohnert** [comment]
Discussion on 09/06/21 between myself, @DominicOram and @JamesKingWork. The following options were considered:
1. All read only "Outputs" are fully configurable in the script definition
2. Existing "Time Estimation" remains as is, additional Outputs are fully configurable
3. Just "Event Estimation" is added as an additional output, which works in the same manner as "Time Estimation"

We decided against 3 as a way of future proofing the system. 1 looked good, except for the need for displaying a total duration just based off events - this would mean the Java side needs to be aware of parameters from the script definition that affect time estimation (Count Rate in the case of event estimation). We decided this makes the interaction between Java and Python too complex, so we settled on 2 as a compromise, i.e. we assume Time Estimation is a given for which we can display a total, while additional outputs are configurable.

The plan for implementation at a lower level is:
- Python Script definition to provide Outputs as a Dict of keys and callback functions
- Python support module iterates through these and returns a Dict of keys and current values
- Java just displays the given values in the table
